Improve installation instructions in the README

kouroshD opened the pull request:
@GiulioRomualdi  @traversaro  
Can you please check the updates for the dependencies installations, and if it is fine merge it? thanks

traversaro commented on README.md:
```diff
@@ -21,10 +21,128 @@ It implements the following architecture:
+
+   1. `CybSDK_app`: you can download the application from [this link](https://developer.cyberith.com/downloads). Go to 'Tools' tab, download the 'Virtualizer Control Panel'. Copy the download application to the workspace. Everytime start using the virtualizer, please calibrate it using this application.
+     
+   2. `CybSDK`: you can download the application from [this link](https://developer.cyberith.com/downloads). Go to 'SDK' tab, download the 'C++ SDK for Windows (Developed for: Visual Studio 2015 Community)'. Copy the download application to the workspace. in theory, we should build it in our Windows machine, but time being it is already installed and is working, so you do not need to install it.
```
I am not getting the last sentence to be honest. It is already installed were? Consider that this document is a generic installation documentation, so it is not safe to refer to the state of a specific setup.

kouroshD replied:
I have updated the sentence, I was trying to say there is already the '.dll' binary, so they do not need to build it again.

traversaro commented on README.md:
```diff
@@ -21,10 +21,128 @@ It implements the following architecture:
+     
+   2. `CybSDK`: you can download the application from [this link](https://developer.cyberith.com/downloads). Go to 'SDK' tab, download the 'C++ SDK for Windows (Developed for: Visual Studio 2015 Community)'. Copy the download application to the workspace. in theory, we should build it in our Windows machine, but time being it is already installed and is working, so you do not need to install it.
+   
+      - Add the following varibales to the Rapid Evironment Edittor:
```
I would rephrase: 
~~~
Append the following diretory to the `Path` User environmental variable, for example using the [Rapid Environment Editor](https://www.rapidee.com): 
~~~
(but probably it make sense to just mention the Rapid Enviroment Editor once in all the README

kouroshD commented on README.md:
```diff
@@ -21,10 +21,128 @@ It implements the following architecture:
-* [Walking-controllers](https://github.com/robotology/walking-controllers) to handl the hand cartesian task and walking commands
-* [Oculus SDK](https://developer.oculus.com/downloads/package/oculus-sdk-for-windows/): The native Windows SDK.
-* [Cyberith SDK](https://www.cyberith.com/research-development/): To allow the virtualizer module to capture the operator data.
+1. **robotology super-build:** Install the robotology super-build on a Windows machine as described here [Windows installation notes](https://github.com/robotology/robotology-superbuild#windows):
```
I have updated it, mentioning superbuild as optional point, and one can install YARP, as mentioned on its official website.

traversaro commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+
+              `LibOVR_LibOVR_LIBRARY_RELEASE`: '<path to codes workspace>/OculusSDK/LibOVR/Lib/Windows/x64/Release/VS2015/LibOVR.lib'
+
+              `YARP_USE_LOBOVR`: check the box
```
```suggestion
               `YARP_USE_LIBOVR`: check the box
```

traversaro commented:
Ok for me, probably is better to wait for @GiulioRomualdi as the current mantainer for merging the PR.

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
-* [Cyberith SDK](https://www.cyberith.com/research-development/): To allow the virtualizer module to capture the operator data.
+ 1. **robotology superbuild [optional]:** Install the robotology superbuild on a Windows machine as described here [Windows installation notes](https://github.com/robotology/robotology-superbuild#windows):
+
+    1. opencv: Do not use the opencv package as mentioned there (Dec 2018), instead install it from the [opencv website] (https://opencv.org/releases.html), version 3.4.4;
```
Can you explain better ```Do not use the opencv package as mentioned there (Dec 2018)```, You can just add the link at the following issue: https://github.com/robotology/robotology-superbuild/issues/145  
Furthermore there is  there is a space between **]** and **(** ```[opencv website] (https://opencv.org/releases.html)```

kouroshD replied:
done.

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+
+   + if you are not using superbuild, look at [this link](http://www.yarp.it/) for installing YARP.
+
+3.  **Walking-controllers:** In order to teleoperate the robot, this package should be installed in another workstation;
```
Actually, you do not need to install the walking controller inside the windows machine.

kouroshD replied:
yes, you are right, I have updated it.

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+
+3.  **Walking-controllers:** In order to teleoperate the robot, this package should be installed in another workstation;
+
+4. In the Super-build, enable the QT5 and Eigen3 to install (follow the instructions there).
```
Which instructions?

kouroshD replied:
added!

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+
+4. In the Super-build, enable the QT5 and Eigen3 to install (follow the instructions there).
+
+5. **Oculus SDK:** In order to install the oculus SDK, yolu need to install first the Oculus app on windows (oculus application), install the depenecies(GLEW, GLFW3), and finally install the sdk (LibOVR):
```
Typo `yolu` :smile:

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+     
+      - extract and copy the glew library to your robot/code workspace (in our case in same path of robotology-superbuild)
+      
+      - the library has the makfiles, so go to the 'glew\glew-2.1.0\glew-2.1.0\build\vc12' and using the VS15 (VS 2017) build the library (release and debug!)
```
Typo `makfiles`

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+              
+              `YARP_USE_GLEW`: check the box
+         
+      - Add the following values to the given variables of the Rapid Environment Editor:
```
I think you can avoid referring to _Rapid Environment Editor_.

kouroshD replied:
I have updated it, but I kept the software we have used as an option!

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+      - if you have problems with compiling GLFW, [this link](https://www.glfw.org/docs/latest/compile_guide.html#compile_generate) or [here](https://github.com/nigels-com/glew) may help.
+          
+          
+    4. `LibOVR`: to dowload the SDK, you can go to [this website](), choose "Native Windows", then "Core Package: OCULUS SDK for windows". select the version 1.16.0 to download (you can install the latest one as well 1.32 (the last release) but there are some breaking changes in the APIs in version 1.17 and you will probably need to patch the driver software). To download you can go easily to [li ...
```
`LibOVR: to dowload the SDK, you can go to [this website]()`. Link missing. + Typo (dowload).

Since the version 1.32 is not compatible with YARP you can directly write: 

4. `LibOVR`: to download the SDK, you can go to [this website](), choose "Native Windows", then "Core Package: OCULUS SDK for windows" and select the version 1.16.0. Or easily follow this [link](https://developer.oculus.com/downloads/package/oculus-sdk-for-windows/1.16.0/).

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+          
+    4. `LibOVR`: to dowload the SDK, you can go to [this website](), choose "Native Windows", then "Core Package: OCULUS SDK for windows". select the version 1.16.0 to download (you can install the latest one as well 1.32 (the last release) but there are some breaking changes in the APIs in version 1.17 and you will probably need to patch the driver software). To download you can go easily to [li ...
+   
+      - Extract and place the package in your workspace (in our case in same path of robotology-superbuild), 1inside that there are two libraries which we need: LibOVR, and LibOVRKernel.
```
typo `1inside`

kouroshD replied:
so many typos!

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+        
+        '\LibOVR\Projects\Windows\VS2015\LibOVR.vcxproj'
+        
+        '\LibOVRKernel\Projects\Windows\VS2015\LibOVRKernel.vcxproj '
```
Please use \` \` environment when you specify a path (e.g.  '\LibOVR\Projects\Windows\VS2015\LibOVR.vcxproj'
 becomes `\LibOVR\Projects\Windows\VS2015\LibOVR.vcxproj`)

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+     
+   2. `CybSDK`: you can download the application from [this link](https://developer.cyberith.com/downloads). Go to 'SDK' tab, download the 'C++ SDK for Windows (Developed for: Visual Studio 2015 Community)'. Copy the download application to the workspace. in theory, we should build it in our Windows machine, but time being it is already built (there exists the .dll file) and is working, so you do ...
+   
+      - Add the following variables to the Rapid Evironment Editor:
```
Like before you can avoid mentioning `Rapid Evironment Editor`

GiulioRomualdi replied:
Or if you want follow @traversaro suggestion https://github.com/robotology/walking-teleoperation/pull/9#discussion_r243104349

GiulioRomualdi commented on README.md:
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
+
+              `LibOVR_LibOVR_LIBRARY_RELEASE`: '<path to codes workspace>/OculusSDK/LibOVR/Lib/Windows/x64/Release/VS2015/LibOVR.lib'
+
+              `YARP_USE_LOBOVR`: check the box
```
you can remove white lines between   `LibOVR_LibOVR_INCLUDE_DIR:PATH: '<path to codes workspace>/OculusSDK/LibOVR/Include'` and   `LibOVR_LibOVR_LIBRARY_DEBUG: '<path to codes ...` (etc)

kouroshD replied:
I think its readability is higher with a line space in between!

kouroshD commented:
Thank you @GiulioRomualdi for the careful comments. I have updated it. Please have a look again. 
Also, I have added a gitignore for temporary files!

GiulioRomualdi commented:
Well done @kouroshD! We can merge the PR

GiulioRomualdi commented:
rebase and merging
